gapic/schema/wrappers.py

```diff
@@ -1679,6 +1779,51 @@ def with_context(self, *,
             meta=self.meta.with_context(collisions=collisions),
         )
 
+    def build_address_allowlist_for_selective_gapic(self, *,
+                                                    address_allowlist: Set['metadata.Address'],
+                                                    resource_messages: Dict[str, 'MessageType'],
+                                                    services_in_proto: Dict[str, 'Service'],
+                                                    ) -> None:
+        """Builds a set of Addresses of wrapper objects to be included in selective GAPIC generation.
+
+        This method is used to create an allowlist of addresses to be used to filter out unneeded
+        services, methods, messages, and enums at a later step.
+        """
+
+        address_allowlist.add(self.ident)
+
+        if self.lro:
+            self.lro.build_address_allowlist_for_selective_gapic(address_allowlist=address_allowlist,
+                                                                 resource_messages=resource_messages)
+
+        if self.extended_lro:
+            # We need to add the service/method pointed to by self.operation_service to
+            # the allowlist, as it might not have been specified by selective_gapic_generation.
+            # We assume that the operation service lives in the same proto as this one.
```

Comment:
Is this a good assumption? (I honestly don't know.) Will it work with operation mixins?

Reply:
I have to make this assumption, otherwise the lookups won't work. The precedent for this is here: https://github.com/googleapis/gapic-generator-python/blob/e050f4eb3eddfe150f028a2a2bd901899afb965a/gapic/schema/api.py#L1238-L1250